### PR TITLE
feat(user-org) : update `User`  and `Organization` models for Firestore injection

### LINKED ISSUE #200: Complete Firestore user-organization integration
**Description**
Complete Firestore integration for user-organization membership. Persist a user's organization memberships, expose whether a user belongs to any organization, and enable adding memberships from the app.

**Acceptance criteria**
- [ ]  User documents store organization memberships; missing data defaults to none.
- [ ] The model indicates if a user belongs to any organization.
- [ ] The app can read the current user’s organizer state reliably.
- [ ] Adding a membership updates Firestore .
- [ ] Backward compatible with existing users; no crashes on absent data.
- [ ] Tests cover read/write success and error handling.

### PR DESCRIPTION
**Pull request by Bouazz-a**
## Summary
Persist a user’s organization memberships in Firestore, expose a reliable organizer flag, align Profile and Organization Feed with the new backend shape, wire event creation to an organization context, and refresh the test suite for speed and determinism.

**Related issue**
Closes #200 

### Changes
- Persist memberships and expose organizer state
  - `User` stores `organizationIds` with a safe default and a computed `isOrganizer`.
  - `UserRepository` adds `addOrganizationToUser(userId, orgId)` and `removeOrganizationFromUser(userId, orgId)`; keeps `isOrganizer()`.
  - `UserRepositoryFirebase` uses Firestore `FieldValue.arrayUnion`/`FieldValue.arrayRemove` for idempotent writes and computes `isOrganizer()` by reading the current user with null‑safe defaults.
- Profile alignment
  - Map `User.isOrganizer` into `ProfileUiState`.
  - Drive the profile CTA and copy via `UserRepository.isOrganizer()` for immediate UI coherence after membership changes.
- Organization feed completion
  - `OrganizationFeedViewModel` aggregates `getOrganizationsByOwner(userId)` and `getOrganizationsByMember(userId)` into a single list, de‑duplicates by `id`, applies a stable sort, and unifies loading/error/empty handling.
- Become organizer consistency
  - After successful org creation, call `userRepository.addOrganizationToUser(userId, orgId)` so `isOrganizer` flips true instantly across screens.
- Fake repository parity
  - Test double implements `addOrganizationToUser(...)` and `removeOrganizationFromUser(...)` with array semantics to mirror production.
- Event creation with org context
  - Navigation passes `organizationId` to `CreateEventForm`, which loads the organization once via a single `LaunchedEffect(Unit)`, pre‑filling `organizerId`/`organizerName` and blocking submission without a valid org context.
- Tests refresh
  - Add unit tests for `addOrganizationToUser`, `removeOrganizationFromUser`, and `isOrganizer`.
  - Refactor feed tests to the combined owner+member list and verified badges; remove redundant cases and trim timeouts to reduce flakiness and runtime.

### Implementation details
- Firestore membership persistence
  - `arrayUnion`/`arrayRemove` provide de‑duplicated, concurrent‑safe updates. Reads default missing `organizationIds` to an empty list for backward compatibility.
  - `User.isOrganizer` is computed on the model to avoid extra network hops and logic drift.
- Feed aggregation
  - ViewModel combines membership and ownership flows, removes duplicates by `id`, sorts deterministically, and funnels errors into a single error state with `onRetry` calling `refreshOrganizations(userId)`. Loading is driven by flow activity, not ad‑hoc delays.
- Event creation
  - `CreateEventForm` receives `organizationId`, resolves the organization once via `LaunchedEffect(Unit)`, locks organizer fields to that source of truth, and validates before invoking the create action.
- UI reactivity
  - Profile CTA binds to `UserRepository.isOrganizer()`, ensuring immediate updates after `addOrganizationToUser(...)` during org creation.
- Testing reliability
  - Prefer deterministic test tags, direct state assertions, and `waitUntil` on UI signals. Fake repo mirrors array behavior to catch edge cases locally.

### UI/UX
- Profile CTA reflects live organizer status.
- Organization Feed shows a single, complete list of owned and member organizations with verified badges.
- Create Event is prefilled with the selected organization and requires a valid org context.

## Tests
- New: repository tests for `addOrganizationToUser`, `removeOrganizationFromUser`, and `isOrganizer`.
- Updated: Organization Feed UI tests for combined queries, navigation on card click, verified badge rendering, error/retry flow, empty/loading states, and scroll behavior.
- Removed redundant UI cases and reduced timeouts for faster CI.

###Notes
- AI was used for the sole purpose of formatting and writing the *Implementation details* section

### CODE REVIEW

File: app/src/main/java/ch/onepass/onepass/ui/createform/CreateEventFormViewModel.kt [hunk 1]
```diff
@@ -231,25 +238,31 @@ class CreateEventFormViewModel(
...
+
+        _uiState.value = CreateEventUiState.Loading
+
+        val result = eventRepository.createEvent(event)
+        result.fold(
+            onSuccess = { eventId -> _uiState.value = CreateEventUiState.Success(eventId) },
+            onFailure = { error ->
+              _uiState.value = CreateEventUiState.Error(error.message ?: "Failed to create event")
+            })
       }
```
Comment (Zeleph25) [inline app/src/main/java/ch/onepass/onepass/ui/createform/CreateEventFormViewModel.kt]: Very good null safety management!

- You correctly handle the case where `organization` could be null, providing a clear error message.  
- The `formStateToEvent` conversion is also safely handled, with an error state if it fails.  
- Using `_uiState.value = CreateEventUiState.Loading` before calling the repository is a nice touch to indicate progress.  
- The `result.fold` clearly separates success and failure handling, making the code easy to read and maintain.  

Overall, great job managing potential nulls and ensuring robust UI state updates.

**Review by Zeleph25 — APPROVED** [review]
Good job @Bouazz-a !!

- The code changes are well implemented, and the null safety management in the repository and UI logic is very solid.  
- It’s great that you wrote new tests — they cover multiple important scenarios for `UserRepository` methods, including adding, removing, and checking organizations.  
- The tests are clear, thorough, and well-structured.

Overall, very good work — all changes look good, the new tests are meaningful, and everything is well-written. LGTM!

File: app/src/androidTest/java/ch/onepass/onepass/ui/navigation/AppNavHostTest.kt [hunk 2]
```diff
@@ -135,7 +135,8 @@ class AppNavHostTest {
       navController.navigate(NavigationDestinations.Screen.Profile.route)
     }
     composeTestRule.waitForIdle()
-
+    Thread.sleep(500)
+    composeTestRule.waitForIdle()
```
[the remark this answers is not in the document]
Comment (Bouazz-a) [reply app/src/androidTest/java/ch/onepass/onepass/ui/navigation/AppNavHostTest.kt]: the CI timed-out when i only used waitForIdle() once, so i added some more waiting logic
